lib/index.js

```diff
@@ -77,13 +87,46 @@ export default {
     this.subscriptions.dispose();
   },
 
+  fixJob(editor) {
+    let correctScope = false;
+
+    if (editor && editor.buffer && editor.buffer.file && editor.buffer.file.path) {
+      const { path } = editor.buffer.file;
+
+      this.baseScopes.forEach((scope) => {
+        const extension = scope.replace('source', '');
+
+        if (path.lastIndexOf(extension) === path.length - extension.length) {
+          correctScope = true;
+        }
+      });
+    }
+
+    if (!correctScope) {
+      return;
+    }
+
+    const oldFileContent = editor.getText();
+    const cursorPosition = editor.getCursorBufferPosition();
+
+    this.provideLinter().lint(editor, true).then(() => {
+      const newFileContent = editor.getText();
+
+      if (newFileContent !== oldFileContent) {
+        editor.save();
```

Comment:
Instead of saving the file twice, use [TextBuffer::onWillSave](https://atom.io/docs/api/v1.26.1/TextBuffer#instance-onWillSave) to make Atom wait for the save job to complete before saving the file to disk. Note that you will need to make the observation of the `TextEditor`s above this more complex since that method only (publicly) exists on the `TextBuffer`.

Reply:
Saving the file twice? Perhaps that fixes the undo issue I posted about…

Reply:
Yep, this would be the cause of that problem. As the code works currently it is letting Atom save the file, then running the fix job, updating the text, and then saving it again.